src/addons/metaservice/src/main/java/ee/ria/xroad/proxy/serverproxy/RestMetadataServiceHandlerImpl.java

```diff
@@ -240,6 +256,34 @@ private void handleGetOpenApi(ProxyMessage requestProxyMessage) throws IOExcepti
             restResponse.getHeaders().add(new BasicHeader(MimeUtils.HEADER_CONTENT_TYPE,
                     DEFAULT_GETOPENAPI_CONTENT_TYPE));
         }
+
+    }
+
+    private OpenapiDescriptionFiletype getFileType(RestResponse response, String serviceDescriptionURL)
+        throws MalformedURLException {
+        boolean isJson = false;
+        boolean isYaml = false;
+        List<Header> headers = response.getHeaders();
+        for (Header header : headers) {
+            if ("content-type".equalsIgnoreCase(header.getName())) {
+                String contentType = header.getValue();
+                if (contentType.contains("application/json")) {
+                    isJson = true;
+                } else if ("application/x-yaml".equals(contentType)) {
+                    isYaml = true;
+                }
+            }
+        }
+
+        if (isJson) {
+            return OpenapiDescriptionFiletype.JSON;
+        } else if (isYaml) {
+            return OpenapiDescriptionFiletype.YAML;
+        }
+
+        URL url = new URL(serviceDescriptionURL);
```

Comment:
java.net.URL should only be used when one is going to connect a resource represented by the URL (url.openConnection()). Since the "serviceDesciptionURL" is already converted to an URI on line 227, it should be reused (also avoids exception handling here).

Reply:
Changed to URI